Region.py

```diff
@@ -109,6 +109,15 @@ def get_scene(self):
             return None
 
 
+    def change_dungeon(self, new_dungeon):
```

Comment:
Is there a reason for boss rooms to be tagged with a dungeon at all? I think it could be more straightforward to not consider boss rooms as part of any dungeon's hint area, and instead relying on finding the dungeon as the nearest hint area in the same way that interiors do. This would also make hints behave well in mixed pools boss shuffle if that ever becomes a thing.

Reply:
Yes, `can_fill()` in Region.py relies on .dungeon being set to know if a location qualifies for settings like "small keys in any dungeon"

Updating change_dungeon to accept `None` would be a step towards allowing mixed pools, however -- but as this branch doesn't support mixed pools at all it seems a little out-of-scope for the moment.